src/System.Memory/src/System/Span.cs

```diff
@@ -0,0 +1,400 @@
+        /// Thrown when the destination Span is shorter than the source Span.
+        /// </exception>
+        /// </summary>
+        public void CopyTo(Span<T> destination)
+        {
+            if (!TryCopyTo(destination))
+                throw ThrowHelper.CreateThrowArgumentException_DestinationTooShort();
+        }
+
+
+        /// <summary>
+        /// Copies the contents of this span into destination span. If the source
+        /// and destinations overlap, this method behaves as if the original values in 
+        /// a temporary location before the destination is overwritten.
+        /// 
+        /// <returns>If the destination span is shorter than the source span, this method
+        /// return false and no data is written to the destination.</returns>
+        /// </summary>
+        /// <param name="destination">The span to copy items into.</param>
+        public bool TryCopyTo(Span<T> destination)
+        {
+            if ((uint)_length > (uint)destination._length)
+                return false;
+
+            // TODO: This is a tide-over implementation as we plan to add a overlap-safe cpblk-based api to Unsafe. (https://github.com/dotnet/corefx/issues/13427)
+            unsafe
+            {
+                ref T src = ref DangerousGetPinnableReference;
+                ref T dst = ref destination.DangerousGetPinnableReference;
+                IntPtr srcMinusDst = Unsafe.ByteOffset<T>(ref dst, ref src);
+                int length = _length;
+
+                bool srcGreaterThanDst = (sizeof(IntPtr) == sizeof(int)) ? srcMinusDst.ToInt32() >= 0 : srcMinusDst.ToInt64() >= 0;
+                if (srcGreaterThanDst)
+                {
+                    // Source address greater than or equal to destination address. Can do normal copy.
+                    for (int i = 0; i < length; i++)
+                    {
+                        Unsafe.Add<T>(ref dst, i) = Unsafe.Add<T>(ref src, i);
+                    }
+                }
+                else
+                {
+                    // Source address less than destination address. Must do backward copy.
+                    int i = length;
+                    while (i-- != 0)
+                    {
+                        Unsafe.Add<T>(ref dst, i) = Unsafe.Add<T>(ref src, i);
+                    }
+                }
+                return true;
+            }
+        }
+
+        /// <summary>
+        /// Returns true if left and right point at the same memory and have the same length.  Note that
+        /// this does *not* check to see if the *contents* are equal.
+        /// </summary>
+        public static bool operator ==(Span<T> left, Span<T> right)
+        {
+            if (left._length != right._length)
+                return false;
+
+            if (!Unsafe.AreSame<T>(ref left.DangerousGetPinnableReference, ref right.DangerousGetPinnableReference))
+                return false;
+
+            return true;
```

Comment:
I think expression-enabled methods lose their charm once you go past one-liners, but I'll check to see if the && expression generates better code than the if ladder.